src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs

```diff
@@ -221,5 +229,18 @@ private static bool IsInvalidGrantSubError(string subError)
 
             return !s_nonUiSubErrors.Contains(subError);
         }
+
+
+        private static string UpdateExceptionForClaimsChallenge(string message, ApiIds apiId)
```

Comment:
I don't think the exception factory should know about OBO. The OBO flow is better suited to massage the exception message.

Alternatively, provide a general purpose message / link to docs where you can clarify how Claims are handled: 

- in a web app, challenge the user. Ensure the challenge contains the claims, for example by including the Claims string in the authorization URI as a parameter
- in a web api .... 
- in client_credentials and managed identity use WithClaims
- in desktop / mobile apps, call AcquireTokenSilent / AcquireTokenInteractive and change them WithClaims

Reply:
Decided to go with a generic message to remove OBO coupling